app/executor.php

```diff
@@ -395,9 +395,9 @@ function getStorageDevice($root): Device {
 App::post('/v1/execution')
     ->desc('Create an execution')
     ->param('runtimeId', '', new Text(64), 'The runtimeID to execute')
-    ->param('vars', [], new Assoc(), 'Environment variables required for the build', false)
-    ->param('data', '', new Text(8192), 'Data to be forwarded to the function, this is user specified.', true)
-    ->param('timeout', 15, new ValidatorRange(1, 900), 'Function maximum execution time in seconds.', true)
+    ->param('vars', [], new Assoc(), 'Environment variables required for the build')
+    ->param('data', '{}', new Text(8192), 'Data to be forwarded to the function, this is user specified.', true)
+    ->param('timeout', 15, new Range(1, (int) App::getEnv('_APP_FUNCTIONS_TIMEOUT', 900)), 'Function maximum execution time in seconds.')
```

Comment:
Timeout is now a required parameter?

That would be a breaking change for all SDK's.

Reply:
No don't worry about this one. This is the internal API route used for communication b/w appwrite and the executor. The functions controller itself is unchanged .

Reply:
Ah yeah! just realized 😆